README.md
# COVID-19-South-America
**COVID-19 Epidemic in South America** \
**https://bgonzalezbustamante.github.io/COVID-19-South-America/**


CHANGELOG.md
# Changelog
All notable changes to "COVID-19 Epidemic in South America" will be documented in this file.

## [Unreleased]

## [4.0.0] - TBC
### Added
- Proportional Hazards models code.
- Descriptives per people code.
### Changed
- Scripts and folders names standardisation.
- Regional mobility diff-in-diff code.
- Mobility robustness checks.
- Mobility placebo test.
- Revision of upgrade versioning (formats of data sets and preprints).

## [Released]

## [3.16.17] - 2022-09-01
### Changed
- Minor corrections.

## [3.16.16] - 2022-04-01
### Changed
- Minor corrections.

## [3.16.15] - 2022-03-31
### Added
- MIT license on GitHub repository.
- Code of conduct in MD format.
### Changed
- Minor corrections.

## [3.15.15] - 2022-03-12
### Changed
- Minor corrections.

## [3.15.14] - 2021-11-17
### Added
- Repository status badge.
### Changed
- Minor corrections.

## [3.14.14] - 2021-11-15
### Changed
- License in MD format on GitHub repository.
- Revision of upgrade versioning.
- Minor corrections.

## [3.14.13] - 2021-10-31
### Changed
- Minor corrections.

## [3.14.12] - 2021-10-22
### Changed
- Revision of upgrade versioning.
- Changelog in MD format on GitHub repository.

## [3.14.11] - 2021-08-18
### Changed
- Minor corrections.

## [3.14.10] - 2021-05-28
### Added
- Mobility and poverty preprint on private OSF component.
- Replication File: Demonstration for R on private OSF component and GitHub repository.
### Changed
- Mobility robustness checks.
- Mobility placebo test.
- Regional mobility diff-in-diff code.
- Minor corrections.

## [3.13.10] - 2021-05-14
### Added
- ITU mobile-broadband subscriptions​​ indicators backup on GitHub repository.
- ITU Mobile-cellular subscriptions​ indicators backup on GitHub repository.
### Changed
- Mobility robustness checks.
- Mobility placebo test.

## [3.12.10] - 2021-05-13
### Changed
- Mobility robustness checks.
- Mobility placebo test.
- Regional mobility diff-in-diff code.

## [3.12.9] - 2021-05-10
### Added
- Mobility robustness checks.
- Mobility placebo test.
### Changed
- Regional mobility diff-in-diff code.

## [3.11.9] - 2021-05-09
### Changed
- Regional mobility diff-in-diff code.

## [3.11.8] - 2021-05-08
### Changed
- Regional mobility estimations code.
- Poverty sources.

## [3.11.7] - 2021-05-07
### Added
- Oxford COVID-19 Government Response Tracker economic indicators.
### Changed
- Regional mobility estimations code.

## [3.10.7] - 2021-04-16
### Changed
- Regional mobility estimations code.

## [3.10.6] - 2021-04-14
### Added
- Poverty sources.
### Changed
- Regional mobility estimations code.

## [3.9.6] - 2021-04-13
### Added
- Git LFS.
### Changed
- Regional mobility estimations code.

## [3.8.6] - 2021-03-27
### Changed
- Minor corrections.

## [3.8.5] - 2021-03-25
### Changed
- Revision of upgrade versioning.
- Minor corrections.

## [3.8.4] - 2021-03-07
### Changed
- Regional mobility diff-in-diff code.

## [3.8.3] - 2021-03-05
### Added
- Regional mobility diff-in-diff code.

## [3.7.3] - 2021-03-01
### Added
- Google COVID-19 Mobility Data backup on GitHub repository.
- Regional mobility estimations code.

## [3.6.3] - 2021-01-12
### Changed
- Minor corrections.

## [3.6.2] - 2020-12-21
### Changed
- Minor corrections.

## [3.6.1] - 2020-09-25
### Added
- Johns Hopkins University CSSE COVID-19 dataset scraper gist.
- Incidence in South America code gist.
- Parametric Re estimations Colombia code.
- Parametric Re estimations Paraguay code.
- Parametric Re estimations Peru code.
- Parametric Re estimations Uruguay code.

## [3.5.1] - 2020-09-11
### Added
- Replication File: Demonstration for R (DOI: 10.1016/j.worlddev.2020.105180).

## [3.4.1] - 2020-09-10
### Added
- Parametric Re estimations Argentina code.
- Parametric Re estimations Bolivia code.
- Parametric Re estimations Brazil code.
- Parametric Re estimations Chile code.

## [3.3.1] - 2020-09-09
### Changed
- Minor corrections.

## [3.3.0] - 2020-09-01
### Added
- Early projections Paraguay code.
- Early projections Peru code.
- Early projections Uruguay code.

## [3.2.0] - 2020-08-31
### Added
- Early projections precoding.
- Early projections Argentina code.
- Early projections Bolivia code.
- Early projections Brazil code.
- Early projections Chile code.
- Early projections Colombia code.

## [3.1.0] - 2020-08-30
### Added
- Oxford Government Responses Tracker scraper.
- Descriptive analysis code.

## [3.0.0] - 2020-08-29
### Added
- Early responses and demonstration for R section.
- Johns Hopkins CSSE COVID-19 Dataset scraper.
### Changed
- Reorganisation between old online monitor and paper analysis.

## [2.12.19] - 2020-08-26
### Changed
- Minor corrections.

## [2.12.18] - 2020-08-05
### Changed
- Minor corrections.

## [2.12.17] - 2020-07-28
### Changed
- Minor corrections.

## [2.12.16] - 2020-07-20
### Changed
- AM revised on private OSF component.

## [2.12.15] - 2020-07-16
### Changed
- AM revised based on the revision comments on private OSF component.
- Replication File: Demonstration for R on private OSF component.

## [2.12.14] - 2020-07-15
### Added
- New stratified, pooled model.

## [2.11.14] - 2020-07-13
### Changed
- Accepted Manuscript (AM) after the peer review process on private OSF component.

## [2.11.13] - 2020-06-02
### Added
- Early responses preprint on private OSF component.
- Replication File: Demonstration for R on private OSF component.
### Changed
- Minor corrections.

## [2.10.13] - 2020-05-31
### Added
- Our World in Data Dataset backup on GitHub repository.
- V-Dem Core v10 Dataset backup on OSF.

## [2.9.13] - 2020-05-24
### Added
- Stratified, pooled Cox Proportional Hazards Models for suppression interventions.
- Early evolution of the pandemic in South America descriptives.

## [2.8.13] - 2020-05-23
### Changed
- Early projections vs. observed data corrections.
- Oxford COVID-19 Government Response Tracker corrections.
- Burden Index corrections.

## [2.8.12] - 2020-05-20
### Added

mobility/R/functions.R
## +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++

#### Script ID ####

## +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++

## Mobility and COVID-19 in South America Functions
## R version 4.0.2 (2020-06-22) -- "Taking Off Again"
## Date: March-April 2021

## Bastián González-Bustamante (University of Oxford, UK)
## E-mail: bastian.gonzalezbustamante@politics.ox.ac.uk
## https://bgonzalezbustamante.com/

## COVID-19 Pandemic in South America Project
## OSF-Project DOI: 10.17605/OSF.IO/6FM7X
## http://bgonzalezbustamante.github.io/COVID-19-South-America/

## +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++

#### Functions ####

## +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++

## Integer Function
is.integer0 <- function(x)
{
  is.integer(x) && length(x) == 0L
}

## Poverty in Argentina
## GitHub Repository ~/tree/master/data/poverty-sources/ARG
mean_poverty_ARG <- 40.9
f.poverty.ARG <- function(x)
{ if(is.integer0(x)){
  print(NA)
} else if(x == "Buenos Aires"){
  print(17.3)
} else if(x == "Buenos Aires Province"){
  print(47.5)
} else if(x == "Catamarca Province"){
  print(35.2)
} else if(x == "Chaco Province"){
  print(NA)
} else if(x == "Chubut Province"){
  print(NA)
} else if(x == "Cordoba"){
  print(40.7)
} else if(x == "Corrientes"){
  print(41.2)
} else if(x == "Entre Rios"){
  print(NA)
} else if(x == "Formosa Province"){
  print(42.4)
} else if(x == "Jujuy"){
  print(38)
} else if(x == "La Pampa"){
  print(NA)
} else if(x == "La Rioja Province"){
  print(29.8)
} else if(x == "Mendoza Province"){
  print(41.5)
} else if(x == "Misiones Province"){
  print(NA)
} else if(x == "Neuquen"){
  print(37.5)
} else if(x == "Rio Negro"){
  print(NA)
} else if(x == "Salta Province"){
  print(45.5)
} else if(x == "San Juan Province"){
  print(35.8)
} else if(x == "San Luis Province"){
  print(39.2)
} else if(x == "Santa Cruz Province"){
  print(NA)
} else if(x == "Santa Fe Province"){
  print(42.6)
} else if(x == "Santiago del Estero Province"){
  print(42.9)
} else if(x == "Tierra del Fuego Province"){
  print(NA)
} else if(x == "Tucuman"){
  print(41.5)
} else {
  print(NA)
}}

## Poverty in Chile
## GitHub Repository ~/tree/master/data/poverty-sources/CHL
mean_poverty_CHL <- 8.6
f.poverty.CHL <- function(x)
{ if(is.integer0(x)){
  print(NA)
} else if(x == "Arica y Parinacota"){
  print(8.4)
} else if(x == "Tarapacá"){
  print(6.4)
} else if(x == "Antofagasta"){
  print(5.1)
} else if(x == "Atacama"){
  print(7.9)
} else if(x == "Coquimbo"){
  print(11.9)
} else if(x == "Valparaíso"){
  print(7.1)
} else if(x == "Santiago Metropolitan Region"){
  print(5.4)
} else if(x == "O'Higgins"){
  print(10.1)
} else if(x == "Maule"){
  print(12.7)
} else if(x == "Ñuble"){
  print(16.1)
} else if(x == "Bio Bio"){
  print(12.3)
} else if(x == "Araucania"){
  print(17.2)
} else if(x == "Los Ríos"){
  print(12.1)
} else if(x == "Los Lagos"){
  print(11.7)
} else if(x == "Aysén"){
  print(4.6)
} else if(x == "Magallanes and Chilean Antarctica"){
  print(2.1)
} else {
  print(NA)
}}

## Multidimensional Poverty in Chile
## GitHub Repository ~/tree/master/data/poverty-sources/CHL
mean_multi_CHL <- 20.7
f.multi.CHL <- function(x)
{ if(is.integer0(x)){
  print(NA)
} else if(x == "Arica y Parinacota"){
  print(21.8)
} else if(x == "Tarapacá"){
  print(24.9)
} else if(x == "Antofagasta"){
  print(16.4)
} else if(x == "Atacama"){
  print(23.2)
} else if(x == "Coquimbo"){
  print(22.6)
} else if(x == "Valparaíso"){
  print(19)
} else if(x == "Santiago Metropolitan Region"){
  print(20)
} else if(x == "O'Higgins"){
  print(18.5)
} else if(x == "Maule"){
  print(22.5)
} else if(x == "Ñuble"){
  print(24.6)
} else if(x == "Bio Bio"){
  print(17.4)
} else if(x == "Araucania"){
  print(28.5)
} else if(x == "Los Ríos"){
  print(22.2)
} else if(x == "Los Lagos"){
  print(25.5)
} else if(x == "Aysén"){
  print(19)
} else if(x == "Magallanes and Chilean Antarctica"){
  print(10.8)
} else {
  print(NA)
}}

## Poverty in Colombia
## GitHub Repository ~/tree/master/data/poverty-sources/COL
mean_poverty_COL <- 35.7
f.poverty.COL <- function(x)
{ if(is.integer0(x)){
  print(NA)
} else if(x == "Amazonas"){
  print(NA)
} else if(x == "Antioquia"){
  print(29.8)
} else if(x == "Arauca"){
  print(NA)
} else if(x == "Atlantico"){
  print(27.3)
} else if(x == "Bogota"){
  print(27.2)
} else if(x == "Bolivar"){
  print(46)
} else if(x == "Boyaca"){
  print(35.7)
} else if(x == "Caldas"){
  print(28.7)
} else if(x == "Caqueta"){
  print(48.8)
} else if(x == "Casanare"){
  print(NA)
} else if(x == "Cauca"){
  print(59.6)
} else if(x == "Cesar"){
  print(51.7)
} else if(x == "Choco"){
  print(68.4)
} else if(x == "Cordoba"){
  print(54.2)
} else if(x == "Cundinamarca"){
  print(20.4)
} else if(x == "Guaviare"){
  print(NA)
} else if(x == "Huila"){
  print(51.2)
} else if(x == "La Guajira"){
  print(61.8)
} else if(x == "Magdalena"){
  print(53.5)
} else if(x == "Meta"){
  print(32.7)
} else if(x == "Narino"){
  print(51)
} else if(x == "North Santander"){
  print(51.9)
} else if(x == "Putamayo"){
  print(NA)
} else if(x == "Risaralda"){
  print(28.7)
} else if(x == "San Andres and Providencia"){
  print(NA)
} else if(x == "Santander"){
  print(31.1)
} else if(x == "Sucre"){
  print(50.3)
} else if(x == "Tolima"){
  print(39.3)
} else if(x == "Valle del Cauca"){
  print(24)
} else if(x == "Vichada"){
  print(NA)
} else {
  print(NA)
}}

## Poverty in Peru
## GitHub Repository ~/tree/master/data/poverty-sources/PER
mean_poverty_PER <- 16.0020707435589
f.poverty.PER <- function(x)
{ if(is.integer0(x)){
  print(NA)
} else if(x == "Amazonas"){
  print(30.1724485103745)
} else if(x == "Ancash"){
  print(17.2014035176195)
} else if(x == "Apurimac"){
  print(11.3483419932819)
} else if(x == "Arequipa"){
  print(10.5929218558695)
} else if(x == "Ayacucho"){
  print(19.1847914434206)
} else if(x == "Cajamarca"){
  print(17.1953111040605)
} else if(x == "Callao Region"){
  print(5.11481472685651)
} else if(x == "Cusco"){
  print(13.546762877841)
} else if(x == "Huancavelica"){
  print(20.5397542865539)
} else if(x == "Huanuco"){
  print(23.8404272441335)
} else if(x == "Ica"){
  print(8.96345334783655)
} else if(x == "Junin"){
  print(23.1032758602094)
} else if(x == "La Libertad"){
  print(10.1432418346883)
} else if(x == "Lambayeque"){
  print(10.2376802573772)
} else if(x == "Lima Region"){
  print(15.4991405118145)
} else if(x == "Loreto"){
  print(52.4256305274022)
} else if(x == "Madre de Dios"){
  print(23.2150761977107)
} else if(x == "Metropolitan Municipality of Lima"){
  print(8.80510784789396)
} else if(x == "Moquegua"){
  print(10.9651994187931)
} else if(x == "Pasco"){
  print(29.896691812666)
} else if(x == "Piura"){
  print(22.3131259897851)
} else if(x == "Puno"){
  print(20.8666595807204)
} else if(x == "San Martin"){
  print(29.0546286229956)
} else if(x == "Tacna"){
  print(7.48157860009444)
} else if(x == "Tumbes"){
  print(20.7392401397941)
} else if(x == "Ucayali"){
  print(43.447542619887)
} else {
  print(NA)
}}


scraper-monitor/JHU-Scraper.R
## +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++

#### Script ID ####

## +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++

## Johns Hopkins University CSSE COVID-19 Dataset Scraper
## R version 3.6.1 (2019-07-05) -- "Action of the Toes"
## Date: March 2020

## Bastián González-Bustamante
## University of Oxford
## E-mail: bastian.gonzalezbustamante@politics.ox.ac.uk
## Website: http://users.ox.ac.uk/~shil5311/

## COVID-19 Pandemic in South America Project
## OSF-Project DOI: 10.17605/OSF.IO/6FM7X
## http://bgonzalezbustamante.github.io/COVID-19-South-America/

## +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++

## Packages
library(tidyverse)
library(lubridate)

## Local Directory
## setwd("")

## Johns Hopkins University CSSE COVID-19 Dataset
JHU_data <- read.csv("https://raw.githubusercontent.com/CSSEGISandData/COVID-19/master/csse_covid_19_data/csse_covid_19_time_series/time_series_19-covid-Confirmed.csv", sep = ",")

## Period Coverage
dd <- as.numeric((today()-1) - as.Date("2020-01-22"))
begin <- ((today()-1)-dd) ## 22 January
today <- (today()-1)
dates <- seq(as.Date(begin), as.Date(today), by="days")
dates

## UNITED KINGDOM
## which(JHU_data[,2] == "United Kingdom")
## Channel Islands, Gibraltar, Caymand Islands excluded
which(JHU_data[,1] == "United Kingdom")
uk <- slice(JHU_data, which(JHU_data[,1] == "United Kingdom"):
              which(JHU_data[,1] == "United Kingdom")) %>%
  pivot_longer(-c(Province.State, Country.Region, Lat, Long),
               names_to = "Date", values_to = "cumulative_cases") %>%
  mutate(incident_cases = c(0, diff(cumulative_cases)))
## view(uk)
uk <- data.frame(uk, dates)
uk$Province.State <- NULL
uk$Country.Region <- NULL
uk$Lat <- NULL
uk$Long <- NULL
uk$Long <- NULL
uk$Date <- NULL
uk$cumulative_cases <- NULL
names(uk)[1] = "GBR"
sum(uk$GBR)

## SPAIN
which(JHU_data[,2] == "Spain")
esp <- slice(JHU_data, which(JHU_data[,2] == "Spain"):
               which(JHU_data[,2] == "Spain")) %>%
  pivot_longer(-c(Province.State, Country.Region, Lat, Long),
               names_to = "Date", values_to = "cumulative_cases") %>%
  mutate(incident_cases = c(0, diff(cumulative_cases)))
## view(esp)
esp <- data.frame(esp, dates)
esp$Province.State <- NULL
esp$Country.Region <- NULL
esp$Lat <- NULL
esp$Long <- NULL
esp$Long <- NULL
esp$Date <- NULL
esp$cumulative_cases <- NULL
names(esp)[1] = "ESP"
sum(esp$ESP)

## ITALY
which(JHU_data[,2] == "Italy")
ita <- slice(JHU_data, which(JHU_data[,2] == "Italy"):
               which(JHU_data[,2] == "Italy")) %>%
  pivot_longer(-c(Province.State, Country.Region, Lat, Long),
               names_to = "Date", values_to = "cumulative_cases") %>%
  mutate(incident_cases = c(0, diff(cumulative_cases)))
## view(ita)
ita <- data.frame(ita, dates)
ita$Province.State <- NULL
ita$Country.Region <- NULL
ita$Lat <- NULL
ita$Long <- NULL
ita$Long <- NULL
ita$Date <- NULL
ita$cumulative_cases <- NULL
names(ita)[1] = "ITA"
sum(ita$ITA)

## ARGENTINA
which(JHU_data[,2] == "Argentina")
arg <- slice(JHU_data, which(JHU_data[,2] == "Argentina"):
               which(JHU_data[,2] == "Argentina")) %>%
  pivot_longer(-c(Province.State, Country.Region, Lat, Long),
               names_to = "Date", values_to = "cumulative_cases") %>%
  mutate(incident_cases = c(0, diff(cumulative_cases)))
## view(arg)
arg <- data.frame(arg, dates)
arg$Province.State <- NULL
arg$Country.Region <- NULL
arg$Lat <- NULL
arg$Long <- NULL
arg$Long <- NULL
arg$Date <- NULL
arg$cumulative_cases <- NULL
names(arg)[1] = "ARG"
sum(arg$ARG)

## BOLIVIA
which(JHU_data[,2] == "Bolivia")
bol <- slice(JHU_data, which(JHU_data[,2] == "Bolivia"):
               which(JHU_data[,2] == "Bolivia")) %>%
  pivot_longer(-c(Province.State, Country.Region, Lat, Long),
               names_to = "Date", values_to = "cumulative_cases") %>%
  mutate(incident_cases = c(0, diff(cumulative_cases)))
## view(bol)
bol <- data.frame(bol, dates)
bol$Province.State <- NULL
bol$Country.Region <- NULL
bol$Lat <- NULL
bol$Long <- NULL
bol$Long <- NULL
bol$Date <- NULL
bol$cumulative_cases <- NULL
names(bol)[1] = "BOL"
sum(bol$BOL)

## BRAZIL
which(JHU_data[,2] == "Brazil")
bra <- slice(JHU_data, which(JHU_data[,2] == "Brazil"):
               which(JHU_data[,2] == "Brazil")) %>%
  pivot_longer(-c(Province.State, Country.Region, Lat, Long),
               names_to = "Date", values_to = "cumulative_cases") %>%
  mutate(incident_cases = c(0, diff(cumulative_cases)))
## view(bra)
bra <- data.frame(bra, dates)
bra$Province.State <- NULL
bra$Country.Region <- NULL
bra$Lat <- NULL
bra$Long <- NULL
bra$Long <- NULL
bra$Date <- NULL
bra$cumulative_cases <- NULL
names(bra)[1] = "BRA"
sum(bra$BRA)

## CHILE
which(JHU_data[,2] == "Chile")
chl <- slice(JHU_data, which(JHU_data[,2] == "Chile"):
               which(JHU_data[,2] == "Chile")) %>%
  pivot_longer(-c(Province.State, Country.Region, Lat, Long),
               names_to = "Date", values_to = "cumulative_cases") %>%
  mutate(incident_cases = c(0, diff(cumulative_cases)))
## view(chl)
chl <- data.frame(chl, dates)
chl$Province.State <- NULL
chl$Country.Region <- NULL
chl$Lat <- NULL
chl$Long <- NULL
chl$Long <- NULL
chl$Date <- NULL
chl$cumulative_cases <- NULL
names(chl)[1] = "CHL"
sum(chl$CHL)

## COLOMBIA
which(JHU_data[,2] == "Colombia")
col <- slice(JHU_data, which(JHU_data[,2] == "Colombia"):
               which(JHU_data[,2] == "Colombia")) %>%
  pivot_longer(-c(Province.State, Country.Region, Lat, Long),
               names_to = "Date", values_to = "cumulative_cases") %>%
  mutate(incident_cases = c(0, diff(cumulative_cases)))
## view(col)
col <- data.frame(col, dates)
col$Province.State <- NULL
col$Country.Region <- NULL
col$Lat <- NULL
col$Long <- NULL
col$Long <- NULL
col$Date <- NULL
col$cumulative_cases <- NULL
names(col)[1] = "COL"
sum(col$COL)

## ECUADOR
which(JHU_data[,2] == "Ecuador")
ecu <- slice(JHU_data, which(JHU_data[,2] == "Ecuador"):
               which(JHU_data[,2] == "Ecuador")) %>%
  pivot_longer(-c(Province.State, Country.Region, Lat, Long),
               names_to = "Date", values_to = "cumulative_cases") %>%
  mutate(incident_cases = c(0, diff(cumulative_cases)))
## view(ecu)
ecu <- data.frame(ecu, dates)
ecu$Province.State <- NULL
ecu$Country.Region <- NULL
ecu$Lat <- NULL
ecu$Long <- NULL
ecu$Long <- NULL
ecu$Date <- NULL
ecu$cumulative_cases <- NULL
names(ecu)[1] = "ECU"
sum(ecu$ECU)

## PARAGUAY
which(JHU_data[,2] == "Paraguay")
pry <- slice(JHU_data, which(JHU_data[,2] == "Paraguay"):
               which(JHU_data[,2] == "Paraguay")) %>%
  pivot_longer(-c(Province.State, Country.Region, Lat, Long),
               names_to = "Date", values_to = "cumulative_cases") %>%
  mutate(incident_cases = c(0, diff(cumulative_cases)))
## view(pry)
pry <- data.frame(pry, dates)
pry$Province.State <- NULL
pry$Country.Region <- NULL
pry$Lat <- NULL
pry$Long <- NULL
pry$Long <- NULL
pry$Date <- NULL
pry$cumulative_cases <- NULL
names(pry)[1] = "PRY"
sum(pry$PRY)

## PERU
which(JHU_data[,2] == "Peru")
per <- slice(JHU_data, which(JHU_data[,2] == "Peru"):
               which(JHU_data[,2] == "Peru")) %>%
  pivot_longer(-c(Province.State, Country.Region, Lat, Long),
               names_to = "Date", values_to = "cumulative_cases") %>%
  mutate(incident_cases = c(0, diff(cumulative_cases)))
## view(per)
per <- data.frame(per, dates)
per$Province.State <- NULL
per$Country.Region <- NULL
per$Lat <- NULL
per$Long <- NULL
per$Long <- NULL
per$Date <- NULL
per$cumulative_cases <- NULL
names(per)[1] = "PER"
sum(per$PER)

## URUGUAY
which(JHU_data[,2] == "Uruguay")
ury <- slice(JHU_data, which(JHU_data[,2] == "Uruguay"):
               which(JHU_data[,2] == "Uruguay")) %>%
  pivot_longer(-c(Province.State, Country.Region, Lat, Long),
               names_to = "Date", values_to = "cumulative_cases") %>%
  mutate(incident_cases = c(0, diff(cumulative_cases)))
## view(ury)
ury <- data.frame(ury, dates)
ury$Province.State <- NULL
ury$Country.Region <- NULL
ury$Lat <- NULL
ury$Long <- NULL
ury$Long <- NULL
ury$Date <- NULL
ury$cumulative_cases <- NULL
names(ury)[1] = "URY"
sum(ury$URY)

## VENEZUELA
which(JHU_data[,2] == "Venezuela")
ven <- slice(JHU_data, which(JHU_data[,2] == "Venezuela"):
               which(JHU_data[,2] == "Venezuela")) %>%
  pivot_longer(-c(Province.State, Country.Region, Lat, Long),
               names_to = "Date", values_to = "cumulative_cases") %>%
  mutate(incident_cases = c(0, diff(cumulative_cases)))
## view(ven)
ven <- data.frame(ven, dates)
ven$Province.State <- NULL
ven$Country.Region <- NULL
ven$Lat <- NULL
ven$Long <- NULL
ven$Long <- NULL
ven$Date <- NULL
ven$cumulative_cases <- NULL
names(ven)[1] = "VEN"
sum(ven$VEN)

## Combine Datasets
covid19 <- merge(uk, esp, by = "dates", all = TRUE)
covid19 <- merge(covid19, ita, by = "dates", all = TRUE)
covid19 <- merge(covid19, arg, by = "dates", all = TRUE)
covid19 <- merge(covid19, bol, by = "dates", all = TRUE)
covid19 <- merge(covid19, bra, by = "dates", all = TRUE)
covid19 <- merge(covid19, chl, by = "dates", all = TRUE)
covid19 <- merge(covid19, col, by = "dates", all = TRUE)
covid19 <- merge(covid19, ecu, by = "dates", all = TRUE)
covid19 <- merge(covid19, pry, by = "dates", all = TRUE)
covid19 <- merge(covid19, per, by = "dates", all = TRUE)
covid19 <- merge(covid19, ury, by = "dates", all = TRUE)
covid19 <- merge(covid19, ven, by = "dates", all = TRUE)

## Slice Data
covid19 <- slice(covid19, 10:nrow(covid19))

## Export CSV
write.csv(covid19,'data_covid19.csv')
